packages/tabs/README.md

```diff
@@ -0,0 +1,37 @@
+# Tabs Name
+
+Tabs documentation
+
+Example Component: 
+```JS
+class Example extends React.Component<{}, Partial<{ active: string }>> {
+  state = { active: "TI1" };
+  handleClick(active) {
+    return () => this.setState({ active });
+  }
+  getContent(id: string) {
+    return {
+      TI1: `This is Tab Name 1 content`,
+      TI2: `This is Tab Name 2 content`
+    }[id];
+  }
+  render() {
+    const { active } = this.state;
+    return (
+      <Tabs selected={active}>
+        {{
```

Comment:
I would like to challenge the API design.

Having an object as a child is not really a conventional React pattern, so I would like to understand why we should go with it. I also think having two distinct things in here shows that the modeling of the tab behavior is not clear.

[React Tabs](https://github.com/reactjs/react-tabs) uses the tabs of a specific kind as children, similar to your work on Table with the Columns. [React Native Tabs](https://github.com/react-native-community/react-native-tab-view) passes them in as a prop, which is also an option IMHO.

Coming from a DX perspective and assuming I would not want the tabs to be part of my routing (which makes sense as we could otherwise also use links) I would suggest an API like this:

```js
<TabableContent>
  <Tab title="A">
	My A content
  </Tab>
  <Tab title="B">
	My B content
  </Tab>
</TabableContent>
```

This would enable me to split up every tab as a separate component, encapsulating it's naming and content in a reusable way. My assumption about the implementation would be that only the content of the currently active tab is being rendered and that the entire selection part is hidden from me as an implementation detail.

What do you think? Which use case am I forgetting about?

Reply:
I agree with Daniel - the children as an object is a little strange. I think the [React Tabs component API](https://github.com/reactjs/react-tabs#basic-example) is really nice.

Reply:
First I want to start of with an article about children compositions patterns [here](https://medium.com/@martin_hotell/react-children-composition-patterns-with-typescript-56dfc8923c64) 

I wanted to let the complete control over which tab is active and which content is shown to the overarching application. So my intend was to skip state in this component. But this is not explaining why I choose for an object as children instead of component like we did with the table. The example you are giving is removing the need for the object since we only have one type of children. Which still works, however we would need to cycle through all children to generate a list for the tabs and to pick the content for the active tab. 

The following could be an example I could agree on, although I want to raise that the pattern of using an object for named slots is not a bad approach, the question is if it is applicable here. For example with the `PageHeader` component the reasoning is a lot better.

```JSX
<TabableContent active="A"> {/* could also be a id*/}
  <Tab title="A">
	My A content
  </Tab>
  <Tab title="B">
	My B content
  </Tab>
</TabableContent>
```

The important question is what do we want to discuss here object as children (named slots) or if named slots are a good pattern to use for the tabs component? In order to move this PR forward I propose that we only discuss the named slot children for tabs as I feel the general pattern discussion is a bigger discussion where we should include more people from the team. 
Another discussion I would like to raise is if we want to keep state inside of the component shouldn't we just use `react-tabs`, the library mentioned by @danielMSchmidt ?

Reply:
> In order to move this PR forward I propose that we only discuss the named slot children for tabs as I feel the general pattern discussion is a bigger discussion where we should include more people from the team.

I'm interested in this discussion. Could you schedule something or add it to the weekly sync?

---

> Another discussion I would like to raise is if we want to keep state inside of the component

I'm sure we could come up a good component API that would let us render tabs without state, but I'm not opposed to letting this component be stateful.

---

> Shouldn't we just use react-tabs, the library mentioned by @DanielMSchmidt ?

I'd be comfortable with that - it's pretty small and allows us to pass class names for styling